src/pybind/mgr/dashboard/controllers/user.py

```diff
@@ -113,6 +123,35 @@ def set(self, username, password=None, name=None, email=None, roles=None,
         mgr.ACCESS_CTRL_DB.save()
         return User._user_to_dict(user)
 
+    @Endpoint('POST')
+    @CreatePermission
+    @UpdatePermission
+    def validate_password(self, password, username=None, old_password=None):
+        """
+        Check if the password meets the password policy.
+        :param password: The password to validate.
+        :param username: The name of the user (optional).
+        :param old_password: The old password (optional).
+        :return: An object with the properties valid, credits and valuation.
+          'credits' contains the password complexity credits and
+          'valuation' the textual summary of the validation.
+        """
+        result = {'valid': False, 'credits': 0, 'valuation': None}
+        try:
+            result['credits'] = validate_password_policy(password, username, old_password)
+            if result['credits'] < 15:
```

Comment:
I would suggesting moving this block into a new method in `password-policy.service.ts` (or adapt the current method, since it is already checking for the `credits` value).
Then you could call it in `user-form.component.ts:88` and `user-password-form.component.ts:83`.
This way we could translate the valuation strings.

Reply:
Hmmm, IMO this is an strong indication that our i18n system is missing something important here. When we do validations in the backend (what is also planned in the future) and forwarding text to the frontend, then it is absolutely important that we can translate text from the backend, too. Currently failed password policies will raise an exception with an explanation text, this text is forwarded to the frontend and displayed below the input field. Those error messages won't and can't be translated in the moment, too. IMO it would not make sense to relocate the valuation of the password policy check to the frontend only to be able to translate it. Imagine, what happens when the REST API endpoint to validate a password against the policy is called without processing the response in the WebUI; the textual valuation of the password is missed. Because of that the valuation MUST be done in the backend.

Reply:
We will surely run into the same situation sooner or later with other things. The processing and validation of user input becomes more and more complex and we can not do every validation in the frontend. Forwarding textual validation information from the backend to the frontend is somewhat important that we need a solution for that. Refactoring this PR will only schedule the generic problem of our i18n implementation.

Reply:
Per our discussion during today's standup, we'll accept that these messages are being returned in English only for the time being (as it's already the case in other places, e.g. the Ceph health output or the configuration settings). Going forward (post-Octopus), the Dashboard backend should return localized strings where applicable - this is tracked in [issue#43252](https://tracker.ceph.com/issues/43252).